eth/tracers/logger/logger.go

```diff
@@ -80,10 +80,10 @@ type StructLog struct {
 type structLogMarshaling struct {
 	Gas         math.HexOrDecimal64
 	GasCost     math.HexOrDecimal64
-	Memory      hexutil.Bytes
-	ReturnData  hexutil.Bytes
-	OpName      string `json:"opName"` // adds call to OpName() in MarshalJSON
-	ErrorString string `json:"error"`  // adds call to ErrorString() in MarshalJSON
+	Memory      hexutil.Bytes `json:",omitempty"`
```

Comment:
```
❯ ./evm --debug --json --code "0x60006000" run                                                                                                                                 <<<
{"pc":0,"op":96,"gas":"0x2540be400","gasCost":"0x3","memory":"0x","memSize":0,"stack":[],"returnData":"0x","depth":1,"refund":0,"opName":"PUSH1"}
{"pc":2,"op":96,"gas":"0x2540be3fd","gasCost":"0x3","memory":"0x","memSize":0,"stack":["0x0"],"returnData":"0x","depth":1,"refund":0,"opName":"PUSH1"}
{"pc":4,"op":0,"gas":"0x2540be3fa","gasCost":"0x0","memory":"0x","memSize":0,"stack":["0x0","0x0"],"returnData":"0x","depth":1,"refund":0,"opName":"STOP"}
```

`memory` and `returnData` still have `0x` when they're empty so they're not really omitted

Reply:
Here's how I want it to work (not sure it does): 
- If you run it with memory enabled, and returndata enabled, then you get `"memory":"0x"` if memory is indeed blank. 
- If you run it with memory disabled, so then we don't even output `"memory"` at all.

Reply:
>  (not sure it does):

**Narrator: And as it doesn't**

Reply:
The type `Bytes` seems to encode `0x` even if the slice is nil

Reply:
> The type Bytes seems to encode 0x even if the slice is nil

Yes.. `omitempty` might just look at the encoded return value, and check if it's empty. We _don't_ want to change the marshaller for `types.Bytes`, so I think we'll have to just pick one behaviour.